scripts/helm/update-chart-version.sh

```diff
@@ -0,0 +1,276 @@
+#!/usr/bin/env bash
+
+# Write output to error output stream.
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die()
+{
+  local _return="${2:-1}"
+  echo_stderr "$1"
+  exit "${_return}"
+}
+
+help() {
+  cat <<EOF
+Usage: $(basename "$0") [OPTIONS]
+
+Options:
+  --dry-run                                 Get the final version without modifying.
+  --branch <branch name>                    Name of the target branch.
+  --tag <tag>                               Release tag.
+  --type <release/develop>                  Which branch to modify.
+  --chart-version <version>                 Version of the current chart.
+  --publish-release                         To modify the charts for a release.
+  --localpv-provisioner-version <version>   LocalPV Provisioner version.
+  --zfs-localpv-version <version>           ZFS LocalPV version.
+  --lvm-localpv-version <version>           LVM LocalPV version.
+  --mayastor-version <version>              Mayastor version.
+
+Examples:
+  $(basename "$0") --branch release/x.y --chart-version 1.2.3 --openebs-crds-version 2.0.1
+EOF
+}
+
+check_tag_is_valid() {
+    local tag="$1"
+    local current_chart_version="$2"
+
+    if [[ "$(semver validate $tag)" != "valid" ]]; then
+      die "Tag is not a valid semver complaint version"
+    fi
+
+    if [[ $current_chart_version != *"-prerelease" ]]; then
+      die "Chart version($current_chart_version) should be a prerelease format to proceed for tag creation flow"
+    fi
+
+    allowed_diff=("" "patch" "prerelease")
+    diff="$(semver diff "$tag" "$current_chart_version")"
+    if ! [[ " ${allowed_diff[*]} " =~ " $diff " ]]; then
+      die "For release/x.y branch the current chart version($current_chart_version)'s X.Y must exactly match X.Y from tag ($tag)"
+    fi
+}
+
+# yq-go eats up blank lines
+# this function gets around that using diff with --ignore-blank-lines
+yq_ibl()
+{
+  set +e
+  diff_out=$(diff -B <(yq '.' "$2") <(yq "$1" "$2"))
+  error=$?
+  if [ "$error" != "0" ] && [ "$error" != "1" ]; then
+    exit "$error"
+  fi
+  if [ -n "$diff_out" ]; then
+    echo "$diff_out" | patch --quiet --no-backup-if-mismatch "$2" -
+  fi
+  set -euo pipefail
+}
```

Comment:
Could use `source "$ROOT_DIR/mayastor/scripts/utils/yaml.sh"` instead and use the yq_ibl from the extensions repo. See the ./scripts/helm/images.sh script on this repo.

Reply:
Well, that may be a smart way to do stuff but I'd not like to over complicate stuff for 12 LOC.